packages/extras/src/lib/components/Sparkles/Sparkles.svelte

```diff
@@ -0,0 +1,98 @@
+<script lang="ts">
+	import { Three, useFrame } from '@threlte/core'
+	import {
+	  BufferAttribute, BufferGeometry, Color, Points, ShaderMaterial, Vector2,
+	  Vector3,
+	  Vector4, type ColorRepresentation
+	} from 'three'
+	import { randFloatSpread } from 'three/src/math/MathUtils'
+	import type { SparklesProperties } from '../../types/components'
+	import { SparkleShader } from './Shader'
+
+	export let count: SparklesProperties['count'] = 100;
+	export let speed: SparklesProperties['speed'] = 1;
+	export let opacity: SparklesProperties['opacity'] = 1;
+	export let color: SparklesProperties['color'] = 100;
+	export let size: SparklesProperties['size']= Math.random();
+	export let scale: SparklesProperties['scale'] = 1;
+	export let noise: SparklesProperties['noise'] = 1;
```

Comment:
Removing the types from "types/components.ts" will render these prop definitions invalid. Can you think of a good solution? Still, this mechanic is also new to me and this PR is kind of an important testbed to think if the most ergonomic way of preparing the types.

What do you think of exporting the props from Sparkles.svelte.d.ts:

```ts
import type { Events, Props, Slots } from '@threlte/core/src/lib/three/types'
import { SvelteComponentTyped } from 'svelte'
import type { ColorRepresentation, Points, Vector3 } from 'three'

export type SparklesProps = Props<Points> & { … }

export default class Sparkles extends SvelteComponentTyped<
  SparklesProps,
  Events<Points>,
  Slots<Points>
> {}
```

and importing them in the component:

```html
<script>
	import type { SparklesProps } from './Sparkles.svelte'

	export let count: SparklesProps['count'] = 1
</script>
```

It *looks* odd and cyclical but essentially it's only types but needs a test drive.

Reply:
Did that. Should I remove the SparklesProperties from types/components.ts since it is not being used now?

Reply:
👍 yep, needs to be removed!

Reply:
Should be up to date now